src/main/java/mate/academy/springboot/datajpa/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.springboot.datajpa.service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final SpecificationManager<Product> specificationManager;
+
+    public ProductServiceImpl(ProductRepository productRepository,
+                              SpecificationManager<Product> specificationManager) {
+        this.productRepository = productRepository;
+        this.specificationManager = specificationManager;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productRepository.getById(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public Product update(Product product) {
+        return null;
+    }
```

Comment:
```suggestion
    @Override
    public Product update(Product product) {
        return null;
    }
```
This is incorrect method.